File: files/CH-014.R
library(tidyverse)
library(readxl)

input = read_excel("files/CH-014.xlsx", range = "B2:D133")
test  = read_excel("files/CH-014.xlsx", range = "K2:K5")

result = input %>%
  mutate(month = month(Date)) %>%
  group_by(Product) %>%
  summarise(nm = n_distinct(month)) %>%
  filter(nm == 12) %>%
  ungroup() %>%
  select(Products = Product)

identical(result, test)
# [1] TRUE

File: files/000-099/014/CH-014.py
import pandas as pd

input_data = pd.read_excel("CH-014.xlsx", usecols="B:D", skiprows=1, nrows=132)
test = pd.read_excel("CH-014.xlsx", usecols="K", skiprows=1, nrows=4)

input_data["Date"] = pd.to_datetime(input_data["Date"])
result = (
    input_data.assign(month=input_data["Date"].dt.month)
    .groupby("Product", as_index=False)["month"]
    .nunique()
)
result = result.loc[result["month"] == 12, ["Product"]].rename(columns={"Product": "Products"})

print(result.equals(test))
